data/test/scenarios/test_synced_state.cfg

```diff
@@ -0,0 +1,90 @@
+# wmllint: no translatables
+
+# Check that wesnoth.current.synced_state works as documented, and that a subset of the standard events match the docs
+# in https://wiki.wesnoth.org/EventWML#Multiplayer_safety
+{GENERIC_UNIT_TEST "test_synced_state" (
+    [event]
+        name = preload
+        [lua]
+            code =<<
+                    unit_test.assert_equal(wesnoth.current.synced_state, 'preload', 'preload should have its own state')
+                >>
+        [/lua]
+    [/event]
+
+    [event]
+        name = start
+        [lua]
+            code =<<
+                    unit_test.assert_equal(wesnoth.current.synced_state, 'synced', 'start should be synchronised')
+                    local result = wesnoth.sync.evaluate_single(
+                        function()
+                            unit_test.assert_equal(wesnoth.current.synced_state, 'local_choice', 'expected value to be local_choice, within synced event')
+                            return { value = wesnoth.current.synced_state }
+                        end)
+                    unit_test.assert_equal(result.value, 'local_choice', 'wrong value returned, within synced event')
+                >>
+        [/lua]
+    [/event]
+
+    [event]
+        name=turn 1
+        [lua]
+            code =<<
+                    unit_test.assert_equal(wesnoth.current.synced_state, 'synced', 'enemies_defeated should be synced')
```

Comment:
Force-pushing to fix the function name here ("turn X" instead of "enemies_defeated")